README.md

```diff
@@ -46,26 +48,338 @@ ember install ember-resources
+ - or move the logic to the component, which solves
+
+This example will move the logic to the component, because eliminating the helper
+also eliminates a few of the downsides mentioned in the previous example.
+
+For re-use, this could be extracted to a _provider_ component.
+
+```js
+// app/components/load-records.js
+import Component from '@glimmer/component';
+import { cached, tracked } from '@glimmer/tracking';
+
+export default class LoadRecords extends Component {
+  // the @cached decorator here is required so that repeat
+  // accesses to either request or records do not unnecessarily
+  // create a new TrackedRequest for the same @endpoint
+  @cached
+  get request() {
+    // A new TrackedRequest is created when @endpoint changes
+    let trackedRequest = new TrackedRequest(this.args.endpoint);
+
+    return trackedRequest;
+  }
+
+  get records() {
+    return this.request.results ?? [];
+  }
+}
+
+class TrackedRequest {
+  @tracked results;
+  @tracked error;
+
+  constructor(endpoint) {
+    get(endpoint)
+      .then((results) => (this.results = results))
+      .catch((error) => (this.error = error));
+  }
+}
+
+async function get(endpoint) {
+  let response = await fetch(`https://swapi.dev/api/${endpoint}`);
+  let json = await response.json();
+
+  return json.results;
+}
+```
+```hbs
+{{!-- app/components/load-records.hbs --}}
+{{#each this.records as |record|}}
+  {{record.name}}<br>
+{{/each}}
+```
+and usage of this component would look like:
+```hbs
+<LoadRecords @endpoint="starships" />
+```
+
+This is a huge improvement over the previous example, but still has downsides:
+ - You have to manage state of the request yourself
+ - Still feels like a lot of ceremony for something that _"should be easy"_.
+ - While `TrackedRequest` can be used in JavaScript, requiring a `@cached` getter
+   to use it also feels like too much ceremony.
+
+</details>
+
+<details><summary>With Resources</summary>
+
+This example uses `trackedFunction`, a light API wrapping a plain `Resource`.
+
+```js
+// app/components/load-records.js
+import Component from '@glimmer/component';
+import { trackedFunction } from 'ember-resources';
+
+export default class LoadRecords extends Component {
+  request = trackedFunction(this, async () => {
+    let response = await fetch(`https://swapi.dev/api/${this.args.endpoint}`);
+    let json = await response.json();
+
+    return json.results;
+  }),
+
+  get records() {
+    return this.request.value ?? [];
+  }
+}
+```
+```hbs
+{{!-- app/components/load-records.hbs --}}
+{{#each this.records as |record|}}
+  {{record.name}}<br>
+{{/each}}
+```
+
+</details>
+
+### Web Permissions
+
+TODO
+
+### Selection
+
+This example, non-async,
+
+internal state management.
+
+Could be written as
+
+```js
+@cached
+get selection() {
+  return new Selection(/* initial args here */)
+}
+```
+
+## API and Usage
 
 ### `useResource`
 
 `useResource` takes either a `Resource` or `LifecycleResource` and an args thunk.
```

Comment:
I'm a bit meh about the explanation of the arguments here. Specifically reading from top to bottom it's hard to spot what is `thunk`. Should we list the options the examples of the argument combinations? 

```js
useResource(this, resource, thunk);
```

Don't know exactly.

Reply:
Yeah, I've been thinking about generating TypeDoc for these so that it's way easier to references the named parts

Reply:
Now using typedoc. Links everywhere